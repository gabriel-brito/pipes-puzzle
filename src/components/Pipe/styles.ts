import styled from 'styled-components'

export const Pipe = styled.span`
  align-items: center;
  background: url(/assets/images/tiles.png);
  cursor: pointer;
  display: flex;
  font-size: 24px;
  height: 60px;
  justify-content: center;
  width: 60px;

  &:hover {
    opacity: 0.7;
    border: 1px solid white;
  }

  &[data-symbol='╸'] {
    background-position: 120px 0px;
  }
  &[data-symbol='╹'] {
    background-position: -60px -60px;
  }
  &[data-symbol='╺'] {
    background-position: 60px 0px;
  }
  &[data-symbol='╻'] {
    background-position: 60px -60px;
  }
  &[data-symbol='━'] {
    background-position: 0px 0px;
  }
  &[data-symbol='┃'] {
    background-position: 0px -60px;
  }
  &[data-symbol='┓'] {
    background-position: 120px 60px;
  }
  &[data-symbol='┛'] {
    background-position: 120px 120px;
  }
  &[data-symbol='┗'] {
    background-position: 60px 120px;
  }
  &[data-symbol='┏'] {
    background-position: 0px 60px;
  }
  &[data-symbol='┣'] {
    background-position: 0px -120px;
  }
  &[data-symbol='┳'] {
    background-position: 60px 60px;
  }
  &[data-symbol='┫'] {
    background-position: 60px -120px;
  }
  &[data-symbol='┻'] {
    background-position: 120px -120px;
  }
  &[data-symbol='╋'] {
    background-position: 0px 120px;
  }
`
