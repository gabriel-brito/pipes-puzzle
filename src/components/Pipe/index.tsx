import * as S from './styles'

type PipeTypes = {
  symbol: string
  rotate: () => void
}

export default function Pipe({ symbol, rotate }: PipeTypes) {
  return <S.Pipe data-symbol={symbol} onClick={rotate} data-testid="pipe" />
}
