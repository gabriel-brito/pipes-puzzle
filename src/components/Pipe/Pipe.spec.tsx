import { render, screen } from '@testing-library/react'
import UserEvent from '@testing-library/user-event'
import 'jest-styled-components'

import Pipe from '.'

const props = {
  symbol: 'xpto',
  rotate: jest.fn()
}

describe('Pipe', () => {
  it('should render properly', () => {
    render(<Pipe {...props} />)

    expect(screen.getByTestId('pipe')).toHaveAttribute('data-symbol', 'xpto')
    expect(screen.getByTestId('pipe')).toHaveStyleRule(
      'background',
      'url(/assets/images/tiles.png)'
    )
    expect(screen.getByTestId('pipe')).toMatchSnapshot()
  })

  it('should call a function when clicked', () => {
    render(<Pipe {...props} />)

    UserEvent.click(screen.getByTestId('pipe'))

    expect(props.rotate).toHaveBeenCalledTimes(1)
  })
})
