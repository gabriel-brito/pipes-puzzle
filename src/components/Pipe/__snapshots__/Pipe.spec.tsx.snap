// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipe should render properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: url(/assets/images/tiles.png);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  height: 60px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 60px;
}

.c0:hover {
  opacity: 0.7;
  border: 1px solid white;
}

.c0[data-symbol='╸'] {
  background-position: 120px 0px;
}

.c0[data-symbol='╹'] {
  background-position: -60px -60px;
}

.c0[data-symbol='╺'] {
  background-position: 60px 0px;
}

.c0[data-symbol='╻'] {
  background-position: 60px -60px;
}

.c0[data-symbol='━'] {
  background-position: 0px 0px;
}

.c0[data-symbol='┃'] {
  background-position: 0px -60px;
}

.c0[data-symbol='┓'] {
  background-position: 120px 60px;
}

.c0[data-symbol='┛'] {
  background-position: 120px 120px;
}

.c0[data-symbol='┗'] {
  background-position: 60px 120px;
}

.c0[data-symbol='┏'] {
  background-position: 0px 60px;
}

.c0[data-symbol='┣'] {
  background-position: 0px -120px;
}

.c0[data-symbol='┳'] {
  background-position: 60px 60px;
}

.c0[data-symbol='┫'] {
  background-position: 60px -120px;
}

.c0[data-symbol='┻'] {
  background-position: 120px -120px;
}

.c0[data-symbol='╋'] {
  background-position: 0px 120px;
}

<span
  class="c0"
  data-symbol="xpto"
  data-testid="pipe"
/>
`;
